### PR TITLE
[Branch repo-nexus-client]

### PR DESCRIPTION
**Pull request by heroldus**
### Repository subsystem does not work for group repositories

The code in branch [repo-nexus-client](https://github.com/sonatype/nexus/tree/repo-nexus-client) does not work for group repositories, because **RepositoryGroupResourceResponse** can't be casted to **RepositoryResourceResponse** in [JerseyRepositorySupport](https://github.com/sonatype/nexus/tree/repo-nexus-client/nexus/nexus-client-core/src/main/java/org/sonatype/nexus/client/internal/rest/jersey/subsystem/repository/JerseyRepositorySupport.java).
The fix introduces a new abstract method to override the response class.
### Extend ArtifactMaven subsystem to support upload artifact

Not finished yet, but good starting point

### CODE REVIEW

File: nexus/nexus-client-core/src/main/java/org/sonatype/nexus/client/core/subsystem/repository/GenericRepository.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.client.core.subsystem.repository;
+
+import org.sonatype.nexus.rest.model.RepositoryResource;
+
+/**
+ * TODO
```
Comment (peterlynch) [inline nexus/nexus-client-core/src/main/java/org/sonatype/nexus/client/core/subsystem/repository/GenericRepository.java:18]: Docs?

**Comment by peterlynch** [comment]
Don't know enough about the low level stuff to comment on some of this, but generally there are code formatting issues ( use code format defined in https://github.com/sonatype/nexus/blob/master/sonatype-nexus-eclipse-formatter.xml ), documentation issues  on interfaces especially ( many TODO ), and appears some test coverage lacking although haven't fully confirmed.

**Comment by heroldus** [comment]
Hi peter,

the code in general is written by Alin and not finished yet. That's why I wanted to merge into branch repo-nexus-client. I justed fixed an issue with group repositories.
Thanks for the eclipse formatter link.

**Comment by adreghiciu** [comment]
I will close this pull request as there is already a new repository client: https://github.com/sonatype/nexus/commit/cc3105324661a2ca7a3d9513f447f4c89c41ee95

**Pull request closed without merging**